debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -638,7 +640,48 @@ protected void handleUpdateTableMetadata(Event event) {
         if (recordMakers.assign(tableNumber, tableId)) {
             logger.debug("Received update table metadata event: {}", event);
         } else {
-            logger.debug("Skipping update table metadata event: {}", event);
+            if (context.dbSchema().isTableMonitored(tableId)) {
+                EventHeaderV4 eventHeader = event.getHeader();
+
+                if (inconsistentSchemaHandlingMode == EventProcessingFailureHandlingMode.FAIL) {
+                    logger.error(
+                            "Schema inconsistency detected while processing binlog event at offset {}.{}" +
+                            "Use the mysqlbinlog tool to view the problematic event: mysqlbinlog --start-position={} --stop-position={} --verbose {}",
+                            source.offset(),
+                            System.lineSeparator(),
+                            eventHeader.getPosition(),
+                            eventHeader.getNextPosition(),
+                            source.binlogFilename()
+                    );
+                    throw new ConnectException("Inconsistency in internal schema detected");
+                } else if (inconsistentSchemaHandlingMode == EventProcessingFailureHandlingMode.WARN) {
+                    logger.warn(
+                            "Schema inconsistency detected while processing binlog event at offset {}.{}" +
+                            "This exception will be ignored and the event be skipped.{}" +
```

Comment:
Technically, there's no exception.

"Encountered change event for table whose schema isn't known to this connector. One possible cause is an incomplete database history topic. Take a new snapshot in this case. The event will be ignored".